### DISCUSSION TITLE
Dockerfile: Expand variables if encountered

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#24409] [Suggest an Idea]
### Tell us more.

We make extensive use of Bitbucket Workspace variables to assist with maintainability of certain key variables.

When used within a `bitbucket-pipelines.yml` `pipe` statement, the following error is thrown.

**Example:**
```
- pipe: docker://$REGISTRY_URL/$REGISTRY_ORG/some-pipe:1.2.3
```

```
{
  "skipReason": "invalid-name",
  "replaceString": "$REGISTRY_URL/$REGISTRY_ORG/some-pipe:1.2.3",
  "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
  "datasource": "docker",
  "depType": "docker",
  "updates": []
}
```

Ideally, it would be great if any variables are evaluated/expanded

**Comment by setchy** [comment]
After thinking about this a bit more, this _may_ actually be broader than `manager: bitbucket-pipelines`... 🤔

**Reply by setchy** [reply]
Looks like this section would need enhancing
https://github.com/renovatebot/renovate/blob/c5aae245caad3288fd4e44519c70e8f5c87e7b12/lib/modules/manager/dockerfile/extract.ts#L120-L124

**Comment by viceice** [comment]
you can probably use registry aliases

**Reply by setchy** [reply]
PR: https://github.com/renovatebot/renovate/pull/24422
